src/cartridges/int_adyen_SFRA/cartridge/client/default/js/adyen_checkout/renderGiftcardComponent.js

```diff
@@ -0,0 +1,352 @@
+  const { giftCardSelect, giftCardUl } = getGiftCardElements();
+  if (giftCardUl?.innerHTML) {
+    return;
+  }
+
+  const { paymentMethodsResponse } = store.checkout;
+  const { imagePath } = store.checkoutConfiguration.session;
+
+  const giftCardBrands = paymentMethodsResponse.paymentMethods.filter(
+    (pm) => pm.type === constants.GIFTCARD,
+  );
+
+  giftCardBrands.forEach((giftCard) => {
+    const newListItem = document.createElement('li');
+    newListItem.setAttribute('data-brand', giftCard.brand);
+    newListItem.setAttribute('data-name', giftCard.name);
+    newListItem.setAttribute('data-type', giftCard.type);
+
+    const span = document.createElement('span');
+    span.textContent = giftCard.name;
+    const img = document.createElement('img');
+    img.src = `${imagePath}${giftCard.brand}.png`;
+    img.width = 40;
+    img.height = 26;
+
+    newListItem.appendChild(span);
+    newListItem.appendChild(img);
+
+    giftCardUl.appendChild(newListItem);
+
+    const newOption = document.createElement('option');
+    newOption.textContent = giftCard.name;
+    newOption.value = giftCard.brand;
+    newOption.style.visibility = 'hidden';
+    giftCardSelect.appendChild(newOption);
+  });
+
+  attachGiftCardFormListeners();
+}
+
+function removeGiftCard() {
+  $.ajax({
+    type: 'POST',
+    url: 'Adyen-CancelPartialPaymentOrder',
+    data: JSON.stringify(store.partialPaymentsOrderObj),
+    contentType: 'application/json; charset=utf-8',
+    async: false,
+    success(res) {
+      const adyenPartialPaymentsOrder = document.querySelector(
+        '#adyenPartialPaymentsOrder',
+      );
+
+      const {
+        giftCardsList,
+        giftCardAddButton,
+        giftCardSelect,
+        giftCardUl,
+      } = getGiftCardElements();
+
+      adyenPartialPaymentsOrder.value = null;
+      giftCardsList.innerHTML = '';
+      giftCardAddButton.style.display = 'block';
+      giftCardSelect.value = null;
+      giftCardUl.innerHTML = '';
+
+      store.partialPaymentsOrderObj = null;
+      window.sessionStorage.removeItem(constants.GIFTCARD_DATA_ADDED);
+
+      if (res.resultCode === constants.RECEIVED) {
+        document.querySelector('#cancelGiftCardContainer')?.parentNode.remove();
+        store.componentsObj?.giftcard?.node.unmount('component_giftcard');
+      }
+    },
+  });
+}
+
+function renderAddedGiftCard() {
+  const giftCardData = store.partialPaymentsOrderObj.giftcard;
+  const { imagePath } = store.checkoutConfiguration.session;
+
+  const { giftCardsList, giftCardAddButton } = getGiftCardElements();
+
+  const giftCardDiv = document.createElement('div');
+  giftCardDiv.classList.add('gift-card');
+
+  const brandContainer = document.createElement('div');
+  brandContainer.classList.add('brand-container');
+
+  const giftCardImg = document.createElement('img');
+  const giftCardImgSrc = `${imagePath}${giftCardData.brand}.png`;
+  giftCardImg.setAttribute('src', giftCardImgSrc);
+  giftCardImg.width = 40;
```

Comment:
Can't we use the css class for this instead of hard coding width/height values?

Reply:
This is now resolved :)